### PR TITLE
feat(state): add simple state machine

### PR DESCRIPTION
**Pull request by joshrounsville**
We're using a combination of context and hooks to maintain a simple global state that is a little reduxy. This includes a little bit of middleware to persist our state data to localhost. That part is fairly bare at this point and we need to give some thought to what data we want to persist and when we should clear it, but the outline of an idea is in here.

fetch state:
`const [ state ] = useStateValue();` or `const [{ pieceOfState }] = useStateValue();`

set state:
```
import { useStateValue, actions } from 'state/StateProvider';

function MyComponent() {
  const dispatch = useStateValue()[1]; OR const [state, dispatch] = useStateValue();

  {...more code and stuff}

  const handleSomeAction = () => {
    ...doing some stuff
    dispatch({
      type: actions.ADD_PENDING_FACILITY,
      pendingFacility: { ...data },
    });
  }
}
```

### CODE REVIEW

File: src/pages/facility_confirmation.js [hunk 1]
```diff
@@ -1,48 +1,41 @@
...
 
 import Page from 'components/layouts/Page';
 import PageLoader from 'components/Loader/PageLoader';
 import { FacilityConfirmation as FacilityConfirmationComponent } from 'components/Confirmation';
 
-function FacilityConfirmation({ backend }) {
+function FacilityConfirmation() {
   const history = useHistory();
-  const params = useParams();
+  const [{ pendingFacility }] = useStateValue();
```
Comment (joshrounsville) [inline src/pages/facility_confirmation.js:15]: Maybe `pendingFacility` isn't the right name here. I'm open to ideas!

Comment (davydka) [reply src/pages/facility_confirmation.js:15]: I think it makes sense 👍

**Comment by davydka** [comment]
I think this is great. I've been wanting to try out applying the new useReducer pattern to a global app state for a little while now (rather than just locally to a component), so I'm 100% on board on this. I think the persist state is going to be super useful as well.

**Comment by joshrounsville** [comment]
> I think this is great. I've been wanting to try out applying the new useReducer pattern to a global app state for a little while now (rather than just locally to a component), so I'm 100% on board on this. I think the persist state is going to be super useful as well.

@davydka Great! I definitely think this will need to evolve a bit to be super useful to us, but this is a decent place to start.

**Pull request merged**